crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -202,11 +234,15 @@ def compute_resources_check_errors
 
         {
           compute_resources: {
-            data: [], # TODO: implement cluster health check errors
+            data: compute_resources_status[:errors],
             help: I18n.t("api.upgrade.prechecks.compute_resources_check.help.default")
           }
         }
       end
+
+      def ha_presence_check
```

Comment:
I think you have to either drop this method, as it is not used, or reference it from `ha_presence_status`